onnxruntime/core/framework/sparse_utils.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/allocator.h"
+
+namespace onnxruntime {
+#ifndef SHARED_PROVIDER
+class Tensor;
+class SparseTensor;
+class DataTransferManager;
+namespace common {
+class Status;
+}
+#endif
+
+namespace sparse_utils {
+#if !defined(ORT_MINIMAL_BUILD)
+/// <summary>
+/// This function converts dense tensor into Csr format.
+/// Conversion takes place on CPU. Thus if the source
+/// is not on CPU, a copy is made first. Likewise, if the destination
+/// is not on CPU, the function would perform a copy.
+/// std::string src and destination are assumed to be both on CPU
+/// </summary>
+/// <param name="data_manager"></param>
+/// <param name="src"></param>
+/// <param name="cpu_allocator"></param>
+/// <param name="dst_allocator"></param>
+/// <param name="dst"></param>
+/// <returns></returns>
+Status DenseTensorToSparseCsr(const DataTransferManager& data_manager, const Tensor& src, const AllocatorPtr& cpu_allocator,
```

Comment:
nit: If there's no description for the params is there any point having <param> entries for them?